test/Dialect/FIRRTL/canonicalization.mlir

```diff
@@ -514,4 +513,42 @@ firrtl.module @wire_port_prop1(%in_a: !firrtl.uint<9>, %out_b: !firrtl.flip<uint
   firrtl.connect %out_b, %tmp : !firrtl.flip<uint<9>>, !firrtl.uint<9>
 }
 
+// CHECK-LABEL: @add_cst_prop1
+// CHECK-NEXT:   %c11_ui9 = firrtl.constant(11 : i9) : !firrtl.uint<9>
+// CHECK-NEXT:   firrtl.connect %out_b, %c11_ui9 : !firrtl.flip<uint<9>>, !firrtl.uint<9>
+// CHECK-NEXT:  }
+firrtl.module @add_cst_prop1(%out_b: !firrtl.flip<uint<9>>) {
+  %c6_ui7 = firrtl.constant(6 : ui7) : !firrtl.uint<7>
+  %tmp_a = firrtl.wire : !firrtl.uint<7>
+  %c5_ui8 = firrtl.constant(5 : ui8) : !firrtl.uint<8>
+  firrtl.connect %tmp_a, %c6_ui7 : !firrtl.uint<7>, !firrtl.uint<7>
+  %add = firrtl.add %tmp_a, %c5_ui8 : (!firrtl.uint<7>, !firrtl.uint<8>) -> !firrtl.uint<9>
+  firrtl.connect %out_b, %add : !firrtl.flip<uint<9>>, !firrtl.uint<9>
+}
+
+// CHECK-LABEL: @add_cst_prop2
+// CHECK-NEXT:   %c127_si9 = firrtl.constant(127 : i9) : !firrtl.sint<9>
```

Comment:
I think this constant should be `-1 : i9`, since you constant-fold `-6 + 5`. `-1` and `127` would have an identical 7 bit pattern (`0b111_1111`), but would differ for 9 bits (`0b1_1111_1111` vs `0b0_0111_1111`).